core/models/application.py

```diff
@@ -140,7 +140,7 @@ def shared_with_user(user, is_leader=None):
         elif is_leader == True:
             project_query &= Q(projects__owner__memberships__is_leader=True)
         membership_query = Q(created_by__memberships__group__user=user)
-        return Application.objects.filter(membership_query | project_query | ownership_query).distinct()
+        return Application.objects.filter(query.only_current_apps() | membership_query | project_query | ownership_query).distinct()
```

Comment:
I think you want an AND here. (A | B | C) & current

Reply:
Like line 125 above

Reply:
@cdosborn good point, i'll write up a test-case before requesting review to ensure we get the expected results.